src/tests/src/libs/antares/study/short-term-storage-input/short-term-storage-input-output.cpp

```diff
@@ -451,3 +455,478 @@ BOOST_FIXTURE_TEST_CASE(check_series_save, Fixture)
 }
 
 BOOST_AUTO_TEST_SUITE_END()
+
+BOOST_AUTO_TEST_SUITE(AdditionalConstraintsTests)
+
+BOOST_AUTO_TEST_CASE(Validate_ClusterIdEmpty)
+{
+    ShortTermStorage::AdditionalConstraints constraints;
+    constraints.cluster_id = ""; // Cluster ID is empty
+    constraints.variable = "injection";
+    constraints.operatorType = "less";
+
+    auto [ok, error_msg] = constraints.validate();
+    BOOST_CHECK_EQUAL(ok, false);
+    BOOST_CHECK_EQUAL(error_msg, "Cluster ID is empty.");
+}
+
+BOOST_AUTO_TEST_CASE(Validate_InvalidVariable)
+{
+    ShortTermStorage::AdditionalConstraints constraints;
+    constraints.cluster_id = "ClusterA";
+    constraints.variable = "invalid"; // Invalid variable type
+    constraints.operatorType = "less";
+
+    auto [ok, error_msg] = constraints.validate();
+    BOOST_CHECK_EQUAL(ok, false);
+    BOOST_CHECK_EQUAL(error_msg,
+                      "Invalid variable type. Must be 'injection', 'withdrawal', or 'netting'.");
+}
+
+BOOST_AUTO_TEST_CASE(Validate_InvalidOperatorType)
+{
+    ShortTermStorage::AdditionalConstraints constraints;
+    constraints.cluster_id = "ClusterA";
+    constraints.variable = "injection";
+    constraints.operatorType = "invalid"; // Invalid operator type
+
+    auto [ok, error_msg] = constraints.validate();
+    BOOST_CHECK_EQUAL(ok, false);
+    BOOST_CHECK_EQUAL(error_msg,
+                      "Invalid operator type. Must be 'less', 'equal', or 'greater'.");
+}
+
+BOOST_AUTO_TEST_CASE(Validate_InvalidHours)
```

Comment:
In this test, we mean to test 3 configurations leading to a validation error.
Here we indeed have an error, but : 
- we don't which situation causes the error
- the 2 other configurations may not lead to an error

So we need 3 tests instead of one.